tests/integration/backup/test_v3_backup.py

```diff
@@ -1043,3 +1046,125 @@ def test_gives_non_successful_exit_code_for_legacy_backup_format(
         assert cp.returncode == 1
         assert cp.stderr.decode() == error_message
         assert cp.stdout == b""
+
+
+def test_backup_creation_fails_with_consumer_subscribe_assign(
+    kafka_servers: KafkaServers,
+    producer: KafkaProducer,
+    new_topic: NewTopic,
+    tmp_path: Path,
+) -> None:
+    for i in range(1000):
+        producer.send(
+            new_topic.topic,
+            key=f"message-key-{i}",
+            value=f"message-value-{i}-" + 1000 * "X",
+        )
+    producer.flush()
+
+    backup_location = tmp_path / "fails.log"
+    config = set_config_defaults(
+        {
+            "bootstrap_uri": kafka_servers.bootstrap_servers,
+            "topic_name": new_topic.topic,
+        }
+    )
+
+    class SlowConsumer(KafkaConsumer):
+        def poll(self, *args, **kwargs):
+            # Slow down the consumer so more time is given for `assign` to kick in
+            # This simulates a backup of a _large_ topic
+            time.sleep(0.01)
+            return super().poll(*args, **kwargs)
+
+    class ConsumerContext:
+        def __init__(self):
+            self._consumer = SlowConsumer(
+                bootstrap_servers=kafka_servers.bootstrap_servers,
+                topic=new_topic.topic,
+                enable_auto_commit=False,
+                auto_offset_reset="earliest",
+            )
+            # Something that should not be done normally, the consumer is already
+            # subscribed to the topic
+            # Compare with `test_backup_creation_succeeds_without_consumer_assign`
+            self._consumer.assign([TopicPartition(new_topic.topic, partition=0)])
+
+        def __enter__(self):
+            return self._consumer
+
+        def __exit__(self, exc_type, exc_value, exc_traceback):
+            self._consumer.close()
+
+    with patch("karapace.backup.api._consumer") as consumer_patch:
+        consumer_patch.return_value = ConsumerContext()
+        with pytest.raises(InconsistentOffset):
+            api.create_backup(
+                config=config,
+                backup_location=backup_location,
+                topic_name=api.normalize_topic_name(None, config),
+                version=BackupVersion.V3,
+                replication_factor=1,
+            )
+
+
+def test_backup_creation_succeeds_without_consumer_assign(
```

Comment:
suggestion: Let's add some comment here to make it more obvious why this test is added:

"This test was added to test for regressions after ___ ..."

"The issue facilitates only for large topics because ___"